rotkehlchen/globaldb/handler.py

```diff
@@ -1178,25 +1178,58 @@ def get_historical_price(
         If no price can be found returns None
         """
         querystr = (
-            'SELECT from_asset, to_asset, source_type, timestamp, price FROM price_history '
-            'WHERE from_asset=? AND to_asset=? AND ABS(timestamp - ?) <= ? '
+            'SELECT from_asset, to_asset, source_type, timestamp, '
+            'price, MIN(ABS(timestamp - ?)) FROM price_history '
+            'WHERE from_asset=? AND to_asset=? '
+            'AND timestamp between ? AND ?'
         )
-        querylist = [from_asset.identifier, to_asset.identifier, timestamp, max_seconds_distance]
+        querylist = [timestamp, from_asset.identifier, to_asset.identifier, timestamp - max_seconds_distance, timestamp + max_seconds_distance]  # noqa: E501
         if source is not None:
-            querystr += ' AND source_type=?'
+            querystr += ' AND source_type=? '
             querylist.append(source.serialize_for_db())
 
-        querystr += 'ORDER BY ABS(timestamp - ?) ASC LIMIT 1'
-        querylist.append(timestamp)
-
         with GlobalDBHandler().conn.read_ctx() as cursor:
             query = cursor.execute(querystr, tuple(querylist))
             result = query.fetchone()
-            if result is None:
+            if result[0] is None:
                 return None
 
+        # The result tuple last entry MIN(ABS()) is disregarded in deserialize_from_db
         return HistoricalPrice.deserialize_from_db(result)
 
+    @staticmethod
+    def get_historical_prices(
+            query_data: list[tuple['Asset', 'Asset', Timestamp]],
+            max_seconds_distance: int,
+            source: Optional[HistoricalPriceOracle] = None,
+    ) -> list[Optional['HistoricalPrice']]:
+        """Given a list of from/to/timestamp data to query returns all values
```

Comment:
It will

Reply:
I mean it will/may in the future. I used it from various scripts to save time. Compared to just using the singular one for the queries I made I had 3x speedup.